booking-app/src/main/resources/webapp/js/directives.js

```diff
@@ -573,8 +600,16 @@
                                 scope.form.dto.startTime = rowData.startTime;
                                 scope.form.dto.visitId = extraRowData.visitId;
                                 scope.form.dto.visitBookingDetailsId = extraRowData.visitBookingDetailsId;
-                                scope.form.dto.minDate = scope.parseDate(extraRowData.earliestDate);
-                                scope.form.dto.maxDate = scope.parseDate(extraRowData.latestDate);
+                                scope.form.dto.ignoreDateLimitation = extraRowData.ignoreDateLimitation;
+                                if (!scope.form.dto.ignoreDateLimitation) {
+                                    scope.form.dto.minDate = scope.parseDate(extraRowData.earliestDate);
+                                    scope.form.dto.maxDate = scope.parseDate(extraRowData.latestDate);
+                                } else {
+                                    scope.form.dto.minDate = null;
+                                    scope.form.dto.maxDate = null;
+                                }
+                                scope.earliestDateToReturn = scope.parseDate(extraRowData.earliestDate);
```

Comment:
if you will move it before the if, you don't need to parse the date again inside the if